packages/react-reconciler/src/ReactFiberCommitWork.old.js

```diff
@@ -407,14 +408,30 @@ function commitBeforeMutationEffectsOnFiber(finishedWork: Fiber) {
 
   if ((flags & Snapshot) !== NoFlags) {
     setCurrentDebugFiberInDEV(finishedWork);
+  }
 
-    switch (finishedWork.tag) {
-      case FunctionComponent:
-      case ForwardRef:
-      case SimpleMemoComponent: {
-        break;
+  switch (finishedWork.tag) {
+    case FunctionComponent: {
+      if ((flags & Update) !== NoFlags) {
+        try {
+          commitHookEffectListUnmount(
```

Comment:
There's no clean up needed right? So we don't need this extra traversal. Can leave a comment explaining why it's not there.